src/main/fc/core.c

```diff
@@ -1007,8 +1005,7 @@ void processRxModes(timeUs_t currentTimeUs)
     }
 #endif
 
-    if (IS_RC_MODE_ACTIVE(BOXHORIZON) && canUseHorizonMode) {
-        DISABLE_FLIGHT_MODE(ANGLE_MODE);
+    if (IS_RC_MODE_ACTIVE(BOXHORIZON) && !FLIGHT_MODE(ANGLE_MODE) && sensors(SENSOR_ACC)) {
```

Comment:
Think we should disable ANGLE mode when enabling HORIZON mode unless done elsewhere.

```suggestion
    if (IS_RC_MODE_ACTIVE(BOXHORIZON) && sensors(SENSOR_ACC)) {
        DISABLE_FLIGHT_MODE(ANGLE_MODE);
```

Reply:
Yes. The Angle will more priority then Horizon.
The previous code had same priority logic.

Reply:
Perhaps we only need to add `&& sensors(SENSOR_ACC)` and leave `canUseHorizonMode` intact as it handles other modes depending on being armed / activated.

There are Flight Modes and RC [Box] Modes.

Reply:
I've decreased variables count.
When canUseHorizonMode was set to true, the Angle mode always was turn on. Therefore i checked Angle, instead of canUse.. My code has same result as source without one variable.
But if maybe the source code was better for reading, then i can restore it.

Reply:
canuseHorizon also checks for ALT_HOLD_MODE  and failsafeIsActive()

Reply:
Yes,  if these modes are active with acc, then canuseHorizon set as false, and Angle mode turn on in the next codes rows anyway. Else - canuse is true and Angle is not enabled. Therefore canuse and not Angle is equivalent in code below.

Reply:
And the DISABLE(ANGLE) is undue in Horizon condition, because  angle must be off when we can on horizon

Reply:
Yes, we have much on/off modes logic. The logging of actual modes state is realy necessary.